src/components/AppServicesEmptyState/AppServicesEmptyState.tsx

```diff
@@ -0,0 +1,167 @@
+import React from "react";
+import {
+  Title,
+  Button,
+  EmptyState as PFEmptyState,
+  EmptyStateIcon,
+  EmptyStateBody,
+  TitleSizes,
+  TitleProps,
+  ButtonProps,
+  EmptyStateIconProps,
+  EmptyStateProps as PFEmptyStateProps,
+  EmptyStateBodyProps,
+  ButtonVariant,
+  EmptyStateVariant,
+} from "@patternfly/react-core";
+import PlusCircleIcon from "@patternfly/react-icons/dist/js/icons/plus-circle-icon";
+import SpaceShuttleIcon from "@patternfly/react-icons/dist/js/icons/space-shuttle-icon";
+import LockIcon from "@patternfly/react-icons/dist/js/icons/lock-icon";
+import SearchIcon from "@patternfly/react-icons/dist/js/icons/search-icon";
+import ExclamationCircleIcon from "@patternfly/react-icons/dist/js/icons/exclamation-circle-icon";
+import NotFoundIcon from "../../images/NotFound-Icon.svg";
+import { css } from "@patternfly/react-styles";
+import "./AppServicesEmptyState.css";
+
+export enum AppServicesEmptyStateVariant {
+  GettingStarted = "GettingStarted",
+  NoAccess = "NoAccess",
+  NoResult = "NoResult",
+  NoItems = "NoItems",
+  UnexpectedError = "UnexpectedError",
+  PageNotFound = "PageNotFound",
+}
+
+export type AppServicesEmptyStateProps = {
+  titleProps?: Omit<TitleProps, "children" | "headingLevel"> & {
+    headingLevel?: string;
+  };
+  emptyStateProps?: Omit<PFEmptyStateProps, "children" | "variant"> & {
+    variant?: AppServicesEmptyStateVariant | EmptyStateVariant;
+  };
+  emptyStateIconProps?: EmptyStateIconProps;
+  emptyStateBodyProps?: Omit<EmptyStateBodyProps, "children"> & {
+    body?: string | React.ReactNode;
+  };
+  buttonProps?: Omit<ButtonProps, "children"> & {
+    title?: string;
+    onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+    "data-testid"?: string;
+  };
+  children?: React.ReactNode;
+};
+
+export const AppServicesEmptyState: React.FC<AppServicesEmptyStateProps> = ({
```

Comment:
I really don’t like generic empty state components that take a bunch of props to set it up. It makes it very easy to break it by setting the wrong props, and it's hard to know what's available.

Better having a component for each empty state, with only the props they need (often none).

Check the empty states in the metrics for an example.

Reply:
We need to enhance shared components for better reuse. But at this movements let it be as it is.